core/src/main/java/com/linecorp/armeria/server/encoding/HttpEncoders.java

```diff
@@ -24,80 +24,100 @@
 import com.linecorp.armeria.common.HttpHeaderNames;
 import com.linecorp.armeria.common.RequestHeaders;
 import com.linecorp.armeria.common.annotation.Nullable;
-import com.linecorp.armeria.internal.common.encoding.StreamEncoderFactories;
-import com.linecorp.armeria.internal.common.encoding.StreamEncoderFactory;
-
-import io.netty.handler.codec.compression.Brotli;
+import com.linecorp.armeria.common.encoding.StreamEncoderFactory;
 
 /**
  * Support utilities for dealing with HTTP encoding (e.g., gzip).
  */
 final class HttpEncoders {
-
-    static {
-        // Invoke to load Brotli native binary.
-        Brotli.isAvailable();
-    }
-
     @Nullable
-    static StreamEncoderFactory getEncoderFactory(RequestHeaders headers) {
+    static StreamEncoderFactory determineEncoder(
+            Map<String, StreamEncoderFactory> headerToEncoderFactory, RequestHeaders headers
+    ) {
```

Comment:
Optional) Could be compacted just for general aesthetics (although this is subjective). Ditto for the other locations

```suggestion
            Map<String, StreamEncoderFactory> headerToEncoderFactory, RequestHeaders headers) {
```